docs/guides/gitlab.md

```diff
@@ -0,0 +1,200 @@
+
+- This guide assumes a running instance of Pomerium, already configured with an identity Provider (**IdP**) and running as a docker container on the same host/swarm.
+
+    ::: warning
+    While Pomerium can be configured to use [GitLab as an IdP], we do not recommend doing so while also running it behind Pomerium. In addition to the potential to lock out access to the IdP (breaking access to all routes), we consider it best practice to keep a separation of services between your identity provider protected services, especially those housing sensitive data like source code.
+    :::
+
+- This configuration includes secure communication between Pomerium and GitLab as an upstream service. GitLab's Omnibus configuration uses Nginx to serve the Ruby-based application, which is configured to serve it at the domain name users will access it from. This guide uses [mkcert] to generate a certificate for the upstream service, but this can be adjusted for your in-house certificate solution.
+
+## Configure mkcert
+!!!include(install-mkcert.md)!!!
+
+Create a certificate for the domain that will be used for the GitLab route. For example:
+
+```bash
+mkcert "gitlab.pomerium.localhost.io"
+```
+
+::: tip Note:
+This certificate will only be used by GitLab itself to secure communication from the Pomerium proxy service. The Pomerium configuration determines what certificate is served to the end user.
+:::
+
+## Install GitLab
+
+::: warning Note
+While we do our best to keep our documentation up to date, changes to third-party systems are outside our control. Refer to [GitLab Docker Images] from GitLab's docs as needed, or [let us know](https://github.com/pomerium/pomerium/issues/new?assignees=&labels=&template=bug_report.md) if we need to re-visit this section.
+:::
+
+### Prepare The Environment
+
+1. Configure volumes for persistent data. GitLab suggests defining the environment variable `$GITLAB_HOME` to the root directory for its mounted volumes:
+
+    ```bash
+    export GITLAB_HOME=/srv/gitlab #Adjust the path based on your common Docker volume location.
+    ```
+
+1. In the `$GITLAB_HOME` directory, create three sub-directories: `config`, `data`, and `logs`.
+
+    ```bash
+    mkdir $GITLAB_HOME/{config,data,logs}
+    ```
+
+1. In `$GITLAB_HOME/config` create the directory `ssl`. Move the internal certificate and key, created by mkcert in this example, to the new path:
+
+    ```bash
+    mkdir $GITLAB_HOME/config/ssl
+    mv gitlab.localhost.pomerium.io.pem $GITLAB_HOME/config/ssl/
+    mv gitlab.localhost.pomerium.io.key.pem $GITLAB_HOME/config/ssl/
+    ```
+
+### Install and Configure GitLab
+
+1. Create the docker container. The example command below includes custom configuration options in the `GITLAB_OMNIBUS_CONFIG` variable:
+
+    ```bash
+    docker run --detach \
+      --hostname gitlab-ee \
+      --name gitlab \
+      --restart always \
+      --publish 8443:443 --publish 8080:80 --publish 2022:22 \
```

Comment:
```
      --publish 8443:443 --publish 8080:80 --publish 2022:22 \
```

I think we'd want to expose, not publish those ports. Otherwise external traffic can hit those ports directly afaik.